### PR TITLE
Fix #1644, finish ES misc API functional test

### LINKED ISSUE #1644: Finish functional tests for cFE ES Miscellaneous APIs
**Is your feature request related to a problem? Please describe.**
Need open source functional tests for certifiability

**Describe the solution you'd like**
This function got missed by the existing miscellaneous api test
– CFE_ES_ProcessAsyncEvent - Notification that an asynchronous event was detected by the underlying
OS/PSP.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Requester Info**
Alex Campbell GSFC

### PR DESCRIPTION
**Pull request by nmullane**
**Describe the contribution**
Fixes #1644 

**Testing performed**
1. Build and run cfe functional tests

**Expected behavior changes**
- No impact to behavior

**System(s) tested on**
 - Hardware: PC
 - OS: Ubuntu 20.04

**Contributor Info - All information REQUIRED for consideration of pull request**
Niall Mullane - GSFC 582 Intern

### CODE REVIEW

File: modules/cfe_testcase/src/es_misc_test.c [hunk 1]
```diff
@@ -68,8 +68,15 @@ void TestWriteToSysLog(void)
                "MIR (Manual Inspection Required) for CFE_ES_WriteToSysLog");
 }
 
+void TestProcessAsyncEvent(void)
+{
+    UtPrintf("Testing: CFE_ES_ProcessAsyncEvent");
+    CFE_ES_ProcessAsyncEvent();
```
Comment (skliper) [inline modules/cfe_testcase/src/es_misc_test.c:74]: Wrap in UtAssert_VOIDCALL (depends on nasa/osal#1125) so it gets reported in the log.

**Review by skliper — CHANGES_REQUESTED** [review]

**Review by skliper — APPROVED** [review]

**Comment by nmullane** [comment]
I'll work on this.

**Comment by nmullane** [comment]
This PR depends on https://github.com/nasa/osal/pull/1125

**Comment by nmullane** [comment]
I believe this should be passing now. Is there a way to rerun the github actions checks?

**Comment by skliper** [comment]
Reran and all passed.

**Review by zanzaben — APPROVED** [review]

**Pull request merged**